src/components/App.tsx

```diff
@@ -108,12 +126,51 @@ const initialState: InitialStateProps = {
 };
 
 const App = () => {
-  const { app, menu, movies } = initialState;
+  const { app, genres, menu } = initialState;
+
+  const [selectedMovie, setSelectedMovie] = useState(initialState.selectedMovie);
+  const [movies, setMovies] = useState(initialState.movies);
+
+  const [showModal, setModal] = useState('');
```

Comment:
Usually, enums are used for states with constant values:
```ts
enum ModalState {
 editMovieModal = 'editMovieModal',
 ...
}

const [showModal, setModal] = useState<ModalState|null>(null);
```
And `handleOpenModal` would liik like:
```ts
  const handleOpenModal = (modal: ModalState) => {
    setModal(modal);
  };
```
This will keep your code typesafe